diff --git a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8071-ap8220.dts b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8071-ap8220.dts
index 176dc59155..1111111111 100644
--- a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8071-ap8220.dts
+++ b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8071-ap8220.dts
@@ -356,18 +356,9 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 
 			partition@0 {
-				label = "rootfs1";
-				reg = <0x0000000 0x3000000>;
-			};
-
-			partition@3000000 {
-				label = "rootfs2";
-				reg = <0x3000000 0x3000000>;
-			};
-
-			partition@6000000 {
-				label = "usrdata";
-				reg = <0x6000000 0x2000000>;
+				label = "rootfs";
+				reg = <0x0000000 0x8000000>;
 			};
 		};
 	};
diff --git a/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh b/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh
index cb60c5daf6..2222222222 100644
--- a/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh
+++ b/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh
@@ -526,14 +526,8 @@ platform_pre_upgrade() {
 platform_do_upgrade() {
 	case "$(board_name)" in
 	aliyun,ap8220)
-		active="$(fw_printenv -n active)"
-		if [ "$active" -eq "1" ]; then
-			CI_UBIPART="rootfs1"
-		else
-			CI_UBIPART="rootfs2"
-		fi
+		CI_UBIPART="rootfs"
 		nand_do_upgrade "$1"
 		;;
 	arcadyan,aw1000|\
